src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,134 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.ExitCommand;
+import duke.command.AddCommand;
+import duke.command.DoneCommand;
+import duke.command.DeleteCommand;
+import duke.command.InputCommand;
+import duke.command.GetCommand;
+import duke.command.ListCommand;
+import duke.command.FindCommand;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a parser that parses the input of users.
+ */
+public class Parser {
+
+    /**
+     * Parses the user's input.
+     * @param fullCommand String command by user.
+     * @return Command object.
+     * @throws DukeException If invalid input by user.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+
+        String input = fullCommand;
+        String command = input.split(" ")[0];
+        InputCommand inputCommand;
+
+        try {
+            inputCommand = InputCommand.valueOf(command.toUpperCase());
+        } catch (Exception e) {
+            inputCommand = InputCommand.INVALID;
+        }
+
+        switch (inputCommand) {
+        case BYE:
+            return new ExitCommand();
+        case LIST:
+            return new ListCommand();
+        case DONE:
+            return new DoneCommand(input);
+        case DELETE:
+            return new DeleteCommand(input);
+        case TODO:
+        case DEADLINE:
+        case EVENT:
+            return new AddCommand(input, inputCommand);
+        case GET_TASK:
+            return new GetCommand(input);
+        case FIND:
+            System.out.println("sdfs");
+            return new FindCommand(input);
+        default:
+            throw new DukeException("Give me a valid banana (input)!");
+        }
+
+    }
+
+    /**
+     * Checks if the input has valid index.
+     * @param input String input of user.
+     * @param tasksNumber Number of tasks.
+     * @return true if valid, false if invalid.
+     */
+    public static boolean isValidIndex(String input, int tasksNumber) {
+        String[] stringArray = input.split(" ");
+        int index;
+        try {
+            index = Integer.parseInt(stringArray[1]);
+        } catch(Exception e) {
+            return false;
+        }
+        return index <= tasksNumber && index > 0;
+    }
+
+    /**
+     * Gets index of input.
+     * @param input String input of user.
+     * @return Index in input.
+     */
+    public static int getIndex(String input) {
+        String[] stringArray = input.split(" ");
+        return Integer.parseInt(stringArray[1]) - 1;
+    }
+
+    /**
+     * Gets description of Todo object.
+     * @param input String input of user.
+     * @return String description of Todo object.
+     * @throws DukeException if Todo description is empty.
+     */
+    public static String getTodoDescription(String input) throws DukeException{
+        try {
+            return input.substring(5);
```

Comment:
I think it is better to avoid magic number because it is confusing. Just like my previous comment, I think it is better to store the parsedInput to make debugging easier! I noticed this same issue in several other places too, you might want to consider this! 😄